### PR TITLE
fix/#20498

### PR DESCRIPTION
**Pull request by CodeWithUmair**
## **Description**
This PR is fixes#20498 . Which is to Replace the old Menu Component with new Popover Component.

_1. What is the reason for the change?_
Ans: @deprecated The `<Menu />` component has been deprecated in favor of the new `<Popover>` component from the component-library.

_2. What is the improvement/solution?_
Ans: In this PR, I've changed just `GlobalMenu` component.

## **Manual testing steps**

_1. Step1:_
👉 Open `metamask` extension, Enter your Password to login

_2. Step2:_
👉 Click on `Three Dots` on the `Top Right` side of the header. to see the changes

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before with `Menu Component`**

Menu_Component

### **After with `Popover Component`**

Popover_Component

## **Related issues**

One Issue that I want to tell you is that, You have to close `Popover Component` to click on three dots again.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by CodeWithUmair** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/component-library/popover/popover.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const Popover: PopoverComponent = React.forwardRef(
       referenceElement,
       isOpen,
       title,
-      isPortal = false,
+      isPortal = true,
```
Comment (georgewrmarshall) [inline ui/components/component-library/popover/popover.tsx:40]: Shouldn't make any updates to the defaults if you need to change this prop do it at the implementation stage in `ui/components/multichain/global-menu/global-menu.js`

File: ui/components/component-library/popover/popover.tsx [hunk 2]
```diff
@@ -105,20 +105,21 @@ export const Popover: PopoverComponent = React.forwardRef(
...
           className,
         )}
         ref={ref || setPopperElement}
         {...attributes.popper}
         {...(props as BoxProps<C>)}
-        style={{ ...styles.popper, ...contentStyle, ...props.style }}
+        style={{ ...styles.popper, ...contentStyle, ...props.style}}
       >
         {children}
         {hasArrow && (
```
Comment (georgewrmarshall) [inline ui/components/component-library/popover/popover.tsx:125]: Let's not make any changes to the component file. We should be able to make the necessary overrides in `ui/components/multichain/global-menu/global-menu.js`

File: ui/components/multichain/global-menu/global-menu.js [hunk 3]
```diff
@@ -101,7 +102,7 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
   ///: END:ONLY_INCLUDE_IN
 
   return (
-    <Menu anchorElement={anchorElement} onHide={closeMenu}>
+    <Popover referenceElement={anchorElement} isOpen={closeMenu}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/global-menu/global-menu.js:105]: `isOpen` is a boolean type while `closeMenu` is a function type I think the equivalent prop is `onBlur` for the Popover component

```suggestion
    <Popover referenceElement={anchorElement} onBlur={closeMenu}>
```

File: ui/components/multichain/global-menu/global-menu.js [hunk 4]
```diff
@@ -279,17 +280,17 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+   * Function to close the menu
    */
-  anchorElement: PropTypes.instanceOf(window.Element),
+  closeMenu: PropTypes.func.isRequired,
   /**
-   * Function that closes this menu
+   * The element to which the menu should be anchored
    */
-  closeMenu: PropTypes.func.isRequired,
+  anchorElement: PropTypes.instanceOf(window.Element),
```
Comment (georgewrmarshall) [inline ui/components/multichain/global-menu/global-menu.js:295]: Are there any changes here? It seems they are being changed around? Let's revert these changes if it's just changing the ordering as it makes it difficult to see if there have been any updates.

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @CodeWithUmair, this is a good start. I've left some suggestions. We shouldn't change anything in `ui/components/component-library/popover` unless there is a limitation to the component. To change the styling we should do it at the implementation stage using the props available or as a last resort adding a custom class name.

**Comment by georgewrmarshall** [comment]
It also looks like you have some failing tests that will need to be resolved

**Comment by CodeWithUmair** [comment]
@georgewrmarshall Sir Please review this.

And One thing is that:

image

I have to use this `important` because any of other changes is not applying CSS in `Popover.tsx` Component

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looks like you might have had a bad merge here. There are many files that are being changed that shouldn't.  Would you be able to revert those changes?

**Comment by Ashok1927** [comment]
I don't like as a type of this "co-agent" mail??😢

I like only official email with  "official logo"😢😢

AshokDubey.

Thanks.
please do the needful.

**Pull request closed without merging**